### ISSUE TITLE
name length rule for AppService is more strict than in azure

### ISSUE THREAD

**Issue by johlrich** [pulumi/pulumi-azure#159]
AppService has the same naming restriction as FunctionApp in Azure (1-60 characters, case insensitive, alphanumeric and hyphen) but pulumi will reject it if larger than 24.

`Could not make instance of 'azure:appservice/appService:AppService'.: name 'collab-dir-projector-' plus 8 random chars is longer than maximum length 24`

For some reason AppService isn't in the [naming rules list](https://docs.microsoft.com/en-us/azure/architecture/best-practices/naming-conventions#naming-rules-and-restrictions), but I could confirm they are the same manually by spamming numbers at the end of the name I tried to use until it complained:
![image](https://user-images.githubusercontent.com/1655793/50258247-c41da680-03cc-11e9-8047-3b6d28360790.png)

**Comment by johlrich** [comment]
The problem appears to only comes up if you don't explicitly name something and let the suffixed name  generate. If you explicitly name a resource, it will apply the same rules to that name as terraform currently does, but obviously without the suffix applied (which I personally like). 

This more specific problem exists for a few different resources and not just AppService. I also ran into this with EventHubNamespace and ApplicationInsights generated names.

**Comment by mikhailshilkov** [comment]
This one is sometimes annoying. Where does `24` come from @lukehoban?

**Comment by lukehoban** [comment]
> Where does 24 come from @lukehoban?

This comes from https://github.com/pulumi/pulumi-azure/blob/master/resources.go#L961, which can be overridden on a per-resource basis such as here https://github.com/pulumi/pulumi-azure/blob/master/resources.go#L288.  Unfortunately, many Azure resources do have very low limits on name length, so we had to go with a conservative default.

If we can gain confidence that we've identified all the resources with a low limit, and can override those individually and increase the default, that would be great.
